vue/src/vue-elements/sign-in-btn.vue

```diff
@@ -156,6 +166,72 @@
 			cancelModal: function () {
 				this.$store.state.auth_in_progress = false
 				this.modal.isOpen = false
+			},
+            /**
+             * Add Facebook account.
+             *
+             * @param data Data.
+             */
+			addAccountFB(data) {
+				this.$store.dispatch('fetchAJAXPromise', {
+					req: 'add_account_fb',
+					updateState: false,
+					data: data
+				}).then(response => {
+					window.removeEventListener("message", event => this.getChildWindowMessage(event));
+					this.authPopupWindow.close();
+					window.location.reload();
+				}, error => {
+					this.is_loading = false;
+					Vue.$log.error('Got nothing from server. Prompt user to check internet connection and try again', error)
+				});
+			},
+			getChildWindowMessage: function (event) {
+				if (~event.origin.indexOf(this.appOrigin)) {
+					this.addAccountFB(JSON.parse(event.data));
+					// console.log(event.data);
+				} else {
+					return;
+				}
+			},
+			openPopupFB: function () {
+				let loginUrl = this.appOrigin + this.appPathFB + '?callback_url=' + window.location.origin;
```

Comment:
We need to sign those requests based on the user site. 

Each site can have on plugin install a secret key which will be used to sing the requests to the central app. 
Moreover, we can generate the login URL from the PHP side and localize it as we do here https://github.com/Codeinwp/tweet-old-post/blob/master/includes/admin/class-rop-admin.php#L234 
The callback URL can be generated using `get_admin_url( get_current_blog_id(), 'admin.php?page=TweetOldPost' );`